src/main/java/org/opensearch/knn/index/codec/KNN920Codec/KNN920Codec.java

```diff
@@ -49,4 +65,63 @@ public DocValuesFormat docValuesFormat() {
     public CompoundFormat compoundFormat() {
         return knnFormatFacade.compoundFormat();
     }
+
+    @Override
+    public KnnVectorsFormat knnVectorsFormat() {
+        return perFieldKnnVectorsFormat;
+    }
+
+    /**
+     * Class provides per field format implementation for Lucene Knn vector type
+     */
+    class KNN920PerFieldKnnVectorsFormat extends PerFieldKnnVectorsFormat {
+
+        @Override
+        public KnnVectorsFormat getKnnVectorsFormatForField(final String field) {
+            if (isNotKnnVectorFieldType(field)) {
+                log.debug(
+                    String.format(
+                        "Initialize KNN vector format for field [%s] with default params [max_connections] = \"%d\" and [beam_width] = \"%d\"",
+                        field,
+                        Lucene92HnswVectorsFormat.DEFAULT_MAX_CONN,
+                        Lucene92HnswVectorsFormat.DEFAULT_BEAM_WIDTH
+                    )
+                );
+                return new Lucene92HnswVectorsFormat();
+            }
+            final KNNVectorFieldMapper.KNNVectorFieldType type = (KNNVectorFieldMapper.KNNVectorFieldType) mapperService.get()
```

Comment:
Shouldn't we check mapperService.isPresent() here before calling get?

Reply:
if the code reaches here it has been already checked, as part of the isNotKnnVectorFieldType() call

Reply:
Not sure whether we can leave validation logic to another method and expect that it will be taken care. According to Optional contract, it is always advised to check before call get.

Reply:
I see, I guess it makes sense if someone move that method to another place and we need to make sure our code is safe. Let me update

Reply:
I've added the check, but actually cannot think of something other than throwing exception with meaningful message. Otherwise if will became same as isNotKnnVectorFieldType() method